### PR TITLE
[sdk-51] upgrade react native 0.74.0-rc.6

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Follow up of #27887 for the react-native 0.74 upgrade

# How
update package versions
- react-native 0.74.0.rc.5 -> 0.74.0.rc.6

# Test Plan

- bare-expo ios / android
- expo-go
- fabric-tester ios / android

### CODE REVIEW

File: docs/public/static/diffs/template-bare-minimum/raw/46..47.diff [hunk 1]
```diff
@@ -0,0 +1,535 @@
+diff --git a/templates/expo-template-bare-minimum/android/app/build.gradle b/templates/expo-template-bare-minimum/android/app/build.gradle
```
Comment (gabrieldonadel) [inline docs/public/static/diffs/template-bare-minimum/raw/46..47.diff:1]: Are these changes related to the upgrade?

**Review by gabrieldonadel — COMMENTED** [review]
I think you missed https://github.com/expo/expo/blob/987fee9843edbe762662dcdf316363d31988fbaf/apps/sandbox/package.json#L14

**Review by tsapeta — APPROVED** [review]

File: docs/public/static/diffs/template-bare-minimum/raw/46..47.diff (hunk 1 above)
Comment (Kudo) [reply docs/public/static/diffs/template-bare-minimum/raw/46..47.diff:1]: we could skip generating the diff for each rc bump. just to do it before cuting sdk 51 branch or before sdk 51 release.

**Review by Kudo — APPROVED** [review]
